File: app/src/main/java/ru/application/homemedkit/databaseController/Intake.java
package ru.application.homemedkit.databaseController;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "intakes",
        foreignKeys = @ForeignKey(
                entity = Medicine.class,
                parentColumns = "id",
                childColumns = "medicineId",
                onUpdate = ForeignKey.CASCADE,
                onDelete = ForeignKey.CASCADE))

public class Intake {

    @PrimaryKey(autoGenerate = true)
    public long intakeId;
    public long medicineId;
    public double amount;
    public String interval;
    public String time;
    public String period;
    public String startDate;
    public String finalDate;

    public Intake() {
    }

    @Ignore
    public Intake(long intakeId) {
        this.intakeId = intakeId;
    }

    @Ignore
    public Intake(long medicineId,
                  double amount,
                  String interval,
                  String time,
                  String period,
                  String startDate,
                  String finalDate) {
        this.medicineId = medicineId;
        this.amount = amount;
        this.interval = interval;
        this.time = time;
        this.period = period;
        this.startDate = startDate;
        this.finalDate = finalDate;
    }
}

File: app/src/main/java/ru/application/homemedkit/databaseController/Intake.kt
package ru.application.homemedkit.databaseController

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import ru.application.homemedkit.helpers.BLANK
import java.time.LocalTime

@Entity(
    tableName = "intakes",
    foreignKeys = [ForeignKey(
        entity = Medicine::class,
        parentColumns = arrayOf("id"),
        childColumns = arrayOf("medicineId"),
        onUpdate = CASCADE,
        onDelete = CASCADE
    )]
)
data class Intake(
    @PrimaryKey(autoGenerate = true)
    val intakeId: Long = 0L,
    val medicineId: Long = 0L,
    val amount: Double = 0.0,
    val interval: Int = 0,
    val foodType: Int = -1,
    @TypeConverters(Converters::class)
    val time: List<LocalTime> = emptyList(),
    val period: Int = 0,
    val startDate: String = BLANK,
    val finalDate: String = BLANK
)